subctl/issue-798-show-versions-from-logs.md

```diff
@@ -0,0 +1,106 @@
+this version from the logs.
+
+## Design Details
+
+### Capture version in pod logs
+Add a log message to all submariner pods in the following format:
+
+`component-name version: x.y.z`
+
+The component name will be same as defined in `submariner-operator/pkg/names/names.go` e.g. For Submariner Operator
+running version 0.15.0 this will be `submariner-operator version: 0.15.0`.
+
+Following components already log version but format of log message will change:
+
+* submariner-operator
+
+Following components don't log version and will need changes to log message:
+
+* submariner-gateway
+* submariner-routeagent
+* submariner-lighthouse-agent
+* submariner-lighthouse-coredns
+* submariner-globalnet
+* submariner-metrics-proxy
+* submariner-networkplugin-syncer
+
+### Modify output of show versions
+
+Output of `subctl show versions` will be modified to show version from pod logs as follows:
+
+#### Current output
+
+```
+subctl show versions
+
+Cluster "cluster1"
+ ✓ Showing versions
+COMPONENT                       REPOSITORY           VERSION
+submariner-gateway              quay.io/submariner   0.15.1
+submariner-routeagent           quay.io/submariner   0.15.1
+submariner-metrics-proxy        quay.io/submariner   0.15.1
+submariner-operator             quay.io/submariner   0.15.1
+submariner-lighthouse-agent     quay.io/submariner   0.15.1
+submariner-lighthouse-coredns   quay.io/submariner   0.15.1
+```
+
+#### New output
+```
+subctl show versions
+
+Cluster "cluster1"
+ ✓ Showing versions
+COMPONENT                       REPOSITORY           CONFIURED   RUNNING
+submariner-gateway              quay.io/submariner   0.15.1      0.15.1
+submariner-routeagent           quay.io/submariner   0.15.1      0.15.1
+submariner-metrics-proxy        quay.io/submariner   0.15.1      0.15.1
+submariner-operator             quay.io/submariner   0.15.1      0.15.1
+submariner-lighthouse-agent     quay.io/submariner   0.15.1      0.15.0
+submariner-lighthouse-coredns   quay.io/submariner   0.15.1      0.15.0
+```
+
+### Enhancing diagnose
+
+`subctl diagnose` will also be enhanced to highlight mismatch between configured and running versions. This will not be
+treated as an error as this might be intentional.
+
+### Pros
+* Provides more accurate information to user
+* Easier to troubleshoot issues where actual image version is different from one configured.
+
+### Cons
+* Require changes to all components, not just `subctl`
+
+### Backward Compatibility
+
+* `show version` output will change, which may break any user scripts that rely on output in current format.
+* Submariner Operator already logs the version but format of log message will change. This may also impact
+
+### Alternatives
+
+None
+
+## External Dependencies
+
+None
+
+## User Impact
+
+* Users will need to modify any scripts or tooling that rely on current output of `show versions` or version log message
```

Comment:
We should offer JSON output from `subctl` ;-). (Just a note, not requesting this here!)